From a6e632a1963cac3c565461917466c54ea1df761d Mon Sep 17 00:00:00 2001
From: Xiaowen Liu <b37945@freescale.com>
Date: Sat, 5 Jan 2013 15:32:44 +0800
Subject: [PATCH 05/68] ENGR00215174 change NV21 to NV12 for hardware
 limitation.

make camera service support NV12 other than NV21 for VPU limitation.

Signed-off-by: Xiaowen Liu <b37945@freescale.com>
---
 services/camera/libcameraservice/api1/Camera2Client.cpp      |  1 +
 services/camera/libcameraservice/api1/client2/Parameters.cpp | 10 +++++-----
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/services/camera/libcameraservice/api1/Camera2Client.cpp b/services/camera/libcameraservice/api1/Camera2Client.cpp
index 0b6ca5c..a33f1ad 100644
--- a/services/camera/libcameraservice/api1/Camera2Client.cpp
+++ b/services/camera/libcameraservice/api1/Camera2Client.cpp
@@ -1709,6 +1709,7 @@ size_t Camera2Client::calculateBufferSize(int width, int height,
         case HAL_PIXEL_FORMAT_YCbCr_422_SP: // NV16
             return width * height * 2;
         case HAL_PIXEL_FORMAT_YCrCb_420_SP: // NV21
+        case HAL_PIXEL_FORMAT_YCbCr_420_SP: // NV12
             return width * height * 3 / 2;
         case HAL_PIXEL_FORMAT_YCbCr_422_I: // YUY2
             return width * height * 2;
diff --git a/services/camera/libcameraservice/api1/client2/Parameters.cpp b/services/camera/libcameraservice/api1/client2/Parameters.cpp
index 08af566..75282a7 100644
--- a/services/camera/libcameraservice/api1/client2/Parameters.cpp
+++ b/services/camera/libcameraservice/api1/client2/Parameters.cpp
@@ -112,7 +112,7 @@ status_t Parameters::initialize(const CameraMetadata *info) {
                 supportedPreviewFpsRange);
     }
 
-    previewFormat = HAL_PIXEL_FORMAT_YCrCb_420_SP;
+    previewFormat = HAL_PIXEL_FORMAT_YCbCr_420_SP;
     params.set(CameraParameters::KEY_PREVIEW_FORMAT,
             formatEnumToString(previewFormat)); // NV21
 
@@ -133,7 +133,7 @@ status_t Parameters::initialize(const CameraMetadata *info) {
                 supportedPreviewFormats +=
                     CameraParameters::PIXEL_FORMAT_YUV422SP;
                 break;
-            case HAL_PIXEL_FORMAT_YCrCb_420_SP:
+            case HAL_PIXEL_FORMAT_YCbCr_420_SP:
                 supportedPreviewFormats +=
                     CameraParameters::PIXEL_FORMAT_YUV420SP;
                 break;
@@ -2006,11 +2006,11 @@ const char* Parameters::getStateName(State state) {
 int Parameters::formatStringToEnum(const char *format) {
     return
         !format ?
-            HAL_PIXEL_FORMAT_YCrCb_420_SP :
+            HAL_PIXEL_FORMAT_YCbCr_420_SP :
         !strcmp(format, CameraParameters::PIXEL_FORMAT_YUV422SP) ?
             HAL_PIXEL_FORMAT_YCbCr_422_SP : // NV16
         !strcmp(format, CameraParameters::PIXEL_FORMAT_YUV420SP) ?
-            HAL_PIXEL_FORMAT_YCrCb_420_SP : // NV21
+            HAL_PIXEL_FORMAT_YCbCr_420_SP : // NV21
         !strcmp(format, CameraParameters::PIXEL_FORMAT_YUV422I) ?
             HAL_PIXEL_FORMAT_YCbCr_422_I :  // YUY2
         !strcmp(format, CameraParameters::PIXEL_FORMAT_YUV420P) ?
@@ -2030,7 +2030,7 @@ const char* Parameters::formatEnumToString(int format) {
         case HAL_PIXEL_FORMAT_YCbCr_422_SP: // NV16
             fmt = CameraParameters::PIXEL_FORMAT_YUV422SP;
             break;
-        case HAL_PIXEL_FORMAT_YCrCb_420_SP: // NV21
+        case HAL_PIXEL_FORMAT_YCbCr_420_SP: // NV21
             fmt = CameraParameters::PIXEL_FORMAT_YUV420SP;
             break;
         case HAL_PIXEL_FORMAT_YCbCr_422_I: // YUY2
-- 
1.8.0

